src/Native/Runtime/windows/PalRedhawkMinWin.cpp

```diff
@@ -58,6 +61,28 @@ bool InitializeSystemInfo()
 
 extern bool PalQueryProcessorTopology();
 
+void __stdcall ThreadShutdown();
+
+// Callback that is called when a fiber is detached from a thread
+// Only the home fiber of a thread has the thread pointer stored
+// in the FLS value, others have NULL there.
+void __stdcall FiberDetachCallback(void* lpFlsData)
+{
+    UNREFERENCED_PARAMETER(lpFlsData);
+    ASSERT(g_flsIndex != FLS_OUT_OF_INDEXES);
+    ASSERT(lpFlsData == FlsGetValue(g_flsIndex));
+
+    if (lpFlsData == NULL)
+    {
+        // The current fiber is not the home fiber of a thread, so the thread
+        // is not being shut down.
+        return;
+    }
+
+    // The thread is shutting down
+    ThreadShutdown();
```

Comment:
I think it might be valuable if we were to pass lpFlsData down to ThreadShutdown so that ThreadShutdown can assert that ThreadStore::GetCurrentThread() matches the lpFlsData value.  Is this possible also on unix, or does it not make any sense there?